cli/cli_kafka_runner_test.go

```diff
@@ -0,0 +1,343 @@
+package cli
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/featurebasedb/featurebase/v3/cli/internal"
+	"github.com/featurebasedb/featurebase/v3/cli/kafka"
+	"github.com/featurebasedb/featurebase/v3/errors"
+	"github.com/featurebasedb/featurebase/v3/logger"
+	"github.com/stretchr/testify/require"
+)
+
+// These variables are used throught tests to refer to resouces needed for the
+// tests (i.e. where to find the kafka service)
+var (
+	pilosaHost     string
+	pilosaTLSHost  string
+	pilosaGrpcHost string
+	kafkaHost      string
+	registryHost   string
+	certPath       string
+)
+
+// The variables above are set here. If local is set to true, the tests assume
+// all services are running locally. Otherwise, we use the name of the
+// containers that these tests will have access to in CI.
+func init() {
+	local := true
+	var ok bool
+	if pilosaHost, ok = os.LookupEnv("IDK_TEST_PILOSA_HOST"); !ok {
+		if local {
+			pilosaHost = "localhost:10101"
+		} else {
+			pilosaHost = "pilosa:10101"
+		}
+	}
+	if pilosaTLSHost, ok = os.LookupEnv("IDK_TEST_PILOSA_TLS_HOST"); !ok {
+		pilosaTLSHost = "https://pilosa-tls:10111"
+	}
+	if pilosaGrpcHost, ok = os.LookupEnv("IDK_TEST_PILOSA_GRPC_HOST"); !ok {
+		if local {
+			pilosaGrpcHost = "localhost:20101"
+		} else {
+			pilosaGrpcHost = "pilosa:20101"
+		}
+	}
+	if kafkaHost, ok = os.LookupEnv("IDK_TEST_KAFKA_HOST"); !ok {
+		if local {
+			kafkaHost = "localhost:9092"
+		} else {
+			kafkaHost = "kafka:9092"
+		}
+	}
+	if registryHost, ok = os.LookupEnv("IDK_TEST_REGISTRY_HOST"); !ok {
+		if local {
+			registryHost = "localhost:8081"
+		} else {
+			registryHost = "schema-registry:8081"
+		}
+	}
+	if certPath, ok = os.LookupEnv("IDK_TEST_CERT_PATH"); !ok {
+		certPath = "/certs"
+	}
+}
+
+// Struct used for TestRunner which contains the information needed to ingest
```

Comment:
In general, go doc comments on types should begin with the type name, so in this case it would look like:
```
// KafkaRunnerTest is used for ...
```

Also, I'm guessing this type doesn't need to be exported (i.e. it should be called `kafkaRunnerTest`). As it is now, and with it being in the `cli` package (as opposed to `cli_test`), it's part of the exported, public API.

In general, default to making everything unexported (start with a lowercase letter) first. That includes all the members of the struct.